File: src/cz/cuni/lf1/lge/ThunderSTORM/rendering/ui/EmptyRendererUI.java
package cz.cuni.lf1.lge.ThunderSTORM.rendering.ui;

import cz.cuni.lf1.lge.ThunderSTORM.rendering.IncrementalRenderingMethod;
import javax.swing.JPanel;

public class EmptyRendererUI implements IRendererUI {

  @Override
  public String getName() {
    return "No Renderer";
  }

  @Override
  public JPanel getOptionsPanel() {
    return null;
  }

  @Override
  public void readParameters() {
  }

  @Override
  public void setSize(int width, int height) {
  }

  @Override
  public void recordOptions() {
  }

  @Override
  public void readMacroOptions(String options) {
  }

  @Override
  public int getRepaintFrequency() {
    return 0;
  }

  @Override
  public IncrementalRenderingMethod getImplementation() {
    return null;
  }

  @Override
  public void resetToDefaults() {
  }
}

File: src/main/java/cz/cuni/lf1/lge/ThunderSTORM/rendering/ui/EmptyRendererUI.kt
package cz.cuni.lf1.lge.ThunderSTORM.rendering.ui

import cz.cuni.lf1.lge.ThunderSTORM.estimators.PSF.Molecule
import cz.cuni.lf1.lge.ThunderSTORM.rendering.IncrementalRenderingMethod
import ij.ImagePlus
import javax.swing.JPanel

public class EmptyRendererUI : AbstractRenderingUI() {
    init {
        parameters.loadPrefs()
    }

    public override fun getName()
            = "No Renderer"

    public override fun getOptionsPanel(): JPanel
            = JPanel()

    override fun getMethod(): IncrementalRenderingMethod {
        return object : IncrementalRenderingMethod {
            override fun addToImage(x: DoubleArray?, y: DoubleArray?, z: DoubleArray?, dx: DoubleArray?, dz: DoubleArray?) {}
            override fun addToImage(fits: MutableList<Molecule>?) {}
            override fun getRenderedImage() = ImagePlus()
            override fun getRendererName() = ""
            override fun reset() {}
        }
    }
}
